exemplary_workflow/pyiron/workflow.py

```diff
@@ -0,0 +1,95 @@
+import os
+from pyiron_base import Project
+
+
+# create pyiron project 
+pr = Project("workflow")
+
+# create conda environments for the proprocessing, processing and postprocessing stage 
+for k, v in {
+    "preprocessing": "../source/envs/preprocessing.yaml",
+    "processing": "../source/envs/processing.yaml",
+    "postprocessing": "../source/envs/postprocessing.yaml"
+}.items():
+    pr.conda_environment.create(env_name=k, env_file=v)
+
+
+# Preprocessing
+## generate mesh
+def write_input(input_dict, working_directory):
+    script_name = os.path.join(working_directory, "gmsh.sh")
+    with open(script_name, "w") as f:
+        f.writelines("gmsh -2 -setnumber domain_size " + str(input_dict["domain_size"]) + " unit_square.geo -o square.msh")
+    os.chmod(script_name, 0o744)
+
+gmsh = pr.wrap_executable(
+    job_name="gmsh",
+    executable_str="./gmsh.sh",
+    write_input_funct=write_input,
+    input_dict={"domain_size": 2.0},
+    conda_environment_path=pr.conda_environment.preprocessing,
```

Comment:
Yes, the `conda_environment_path` can be `None` - that is actually the default. For most of our workflows we still use just a single environment. That is also the reason why the `conda_subprocess` package is an optional dependency of `pyiron_base`. In terms of container technology, technically this would be possible but it is not yet implemented. Still the approach would be similar we would create a package that is able to run a subprocess command in a container and then interface this package from `pyiron_base`.